titus-ext/zookeeper/src/main/java/com/netflix/titus/ext/zookeeper/connector/CuratorUtils.java

```diff
@@ -0,0 +1,48 @@
+package com.netflix.titus.ext.zookeeper.connector;
+
+import java.io.IOException;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.data.Stat;
+import rx.Completable;
+
+import static org.apache.zookeeper.KeeperException.Code.OK;
+
+public class CuratorUtils {
+
+    public static boolean createPathIfNotExist(CuratorFramework curator, String fullPath, boolean persisted) {
+        try {
+            Stat pathStat = curator.checkExists().forPath(fullPath);
+            // Create the path only if the path does not exist
+            if (pathStat != null) {
+                return false;
+            }
+            curator.create()
```

Comment:
It is a good question. I am not sure. If an override of existing path will result in an exception, this will be handled by the upper layers. Proper exception handling is required anyway, as the path creation may fail due to other reasons as well.

Yes, the `boolean` result is not used by any caller. As it is a generic utility, I added this result for completeness.